src/Microsoft.AspNetCore.Mvc.ViewFeatures/ViewFeatures/DefaultHtmlGenerator.cs

```diff
@@ -56,7 +55,57 @@ public class DefaultHtmlGenerator : IHtmlGenerator
             IModelMetadataProvider metadataProvider,
             IUrlHelperFactory urlHelperFactory,
             HtmlEncoder htmlEncoder,
-            ClientValidatorCache clientValidatorCache)
+            ClientValidatorCache clientValidatorCache) : this(
+                antiforgery,
+                optionsAccessor,
+                metadataProvider,
+                urlHelperFactory,
+                htmlEncoder,
+                clientValidatorCache,
+                () => new ValidationAttributeProvider(optionsAccessor, metadataProvider, clientValidatorCache))
```

Comment:
Why not have the **old** constructor just do `this(..., new ValidationAttributeProvider(...))`. No hacks needed